YamlValidator/Schema.h

```diff
@@ -109,6 +85,58 @@ class Schema {
 	std::variant<std::shared_ptr<ObjectImplementation>, std::shared_ptr<ArrayImplementation>> schema;
 	Schema YamlToSchema(parser_types::Yaml yaml) {};
 
+	static std::string getTypeName(std::variant<Types, Either, parser_types::YamlValue> instance) {
+		if (std::holds_alternative<parser_types::YamlValue>(instance)) {
+			parser_types::YamlValue yamlValueInstance = std::get<parser_types::YamlValue>(instance);
+
+			if (std::holds_alternative<parser_types::String>(yamlValueInstance))
+				return "String";
+			else if (std::holds_alternative<parser_types::Number>(yamlValueInstance))
+				return "Number";
+			else if (std::holds_alternative<parser_types::Boolean>(yamlValueInstance))
+				return "Boolean";
+			else if (std::holds_alternative<parser_types::Null>(yamlValueInstance))
+				return "Null";
+			else if (std::holds_alternative<parser_types::Timestamp>(yamlValueInstance))
+				return "Timestamp";
+			else if (std::holds_alternative<std::shared_ptr<parser_types::Object>>(yamlValueInstance))
+				return "Object";
+			else if (std::holds_alternative<std::shared_ptr<parser_types::Array>>(yamlValueInstance))
+				return "Array";
+
+			return "";
+		} else if(std::holds_alternative<Either>(instance)) {
+			Either eitherInstance = std::get<Either>(instance);
+
+			std::string name = "Either<";
+
+			for (const Types eitherInstanceType : eitherInstance.values) {
```

Comment:
Tänk på att använda antingen pekare eller referenser när du använder iterator-baserade loopar.

```suggestion
			for (Types& eitherInstanceType : eitherInstance.values) {
```